src/resources/deployment.js

```diff
@@ -71,6 +74,8 @@ function createDeployment (cluster, config) {
     ]
   }
 
+  Object.keys(definition).forEach((key) => (definition[key] == null) && delete definition[key])
```

Comment:
Using `definition[key] = void 0` would be "faster" here

Reply:
validated that this was in fact totally unnecessary, so I just deleted it